sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.students.controller.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  Body,
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  ParseIntPipe,
+  Patch,
+  UnauthorizedException,
+} from "@nestjs/common";
+import {
+  ApiNotFoundResponse,
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from "@nestjs/swagger";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, UserToken } from "../../auth/decorators";
+import { ApiProcessError, ClientTypeBaseRoute } from "../../types";
+import BaseController from "../BaseController";
+import {
+  ApplicationOfferingChangesAPIOutDTO,
+  UpdateApplicationOfferingChangeRequestAPIInDTO,
+} from "./models/application-offering-change-request.dto";
+import { ApplicationOfferingChangeRequestStatus } from "@sims/sims-db";
+import { StudentUserToken } from "../../auth";
+import { STUDENT_UNAUTHORIZED_FOR_APPLICATION_OFFERING_CHANGE_REQUEST } from "../../constants";
+import { ApplicationOfferingChangeRequestControllerService } from "./application-offering-change-request.controller.service";
+import { ApplicationOfferingChangeRequestService } from "../../services";
+
+/**
+ * Application offering change request controller for students client.
+ */
+@AllowAuthorizedParty(AuthorizedParties.student)
+@Controller(
+  "application-offering-change-request/:applicationOfferingChangeRequestId",
+)
+@ApiTags(`${ClientTypeBaseRoute.Student}-application-offering-change-request`)
+export class ApplicationOfferingChangeRequestStudentsController extends BaseController {
+  constructor(
+    private readonly applicationOfferingChangeRequestService: ApplicationOfferingChangeRequestService,
+    private readonly applicationOfferingChangeRequestControllerService: ApplicationOfferingChangeRequestControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Gets the Application Offering Change Request details.
+   * @param applicationOfferingChangeRequestId the Application Offering Change Request id.
+   * @param studentUserToken student user token to authorize the user.
+   * @returns Application Offering Change Request details.
+   */
+  @Get()
+  @ApiNotFoundResponse({
+    description: "Not able to find an Application Offering Change Request.",
+  })
+  async getById(
```

Comment:
I know, the same name is given in the institution too, But I will say, it's better to give a proper method name

Reply:
https://github.com/bcgov/SIMS/pull/2196#discussion_r1297796112

Reply:
I can understand the service level file, But I strongly believe that controllers should be a proper name

Reply:
@dheepak-aot @andrepestana-aot @guru-aot suggestions? If yes, should I update the name of the `getById` method in the `application-offering-change-request.institutions.controller` as well?

Reply:
IMO `getById` sounds more like a generic service operation. Controllers could be more specific like `getApplicationOfferingChangeRequest`